crates/integration/src/lib.rs

```diff
@@ -0,0 +1,554 @@
+                    } else {
+                        String::new()
+                    },
+                    if crate_model.repository.is_some() {
+                        "repository.workspace = true".to_string()
+                    } else {
+                        String::new()
+                    },
+                    if crate_model.license.is_some() {
+                        "license.workspace = true".to_string()
+                    } else {
+                        String::new()
+                    },
+                ),
+            );
+
+            self.add_crate_src_at_path(
+                crate_model,
+                self.temp_dir.path().join("crates").join(&crate_model.name),
+            );
+        }
+    }
+
+    pub fn verify_cargo_project(&self, path: &str) {
+        let path = self.temp_dir.path().join(path);
+
+        let exit_status = std::process::Command::new("cargo")
+            .current_dir(&path)
+            .arg("clippy")
+            .arg("--all-targets")
+            .arg("--")
+            .arg("--deny")
+            .arg("warnings")
+            .stdout(std::process::Stdio::inherit())
+            .stderr(std::process::Stdio::inherit())
+            .spawn()
+            .unwrap()
+            .wait();
+
+        assert!(
+            exit_status.unwrap().success(),
+            "Cargo clippy failed for project at {}",
+            path.display()
+        );
+    }
+
+    pub fn git_status(&self) {
+        let output = std::process::Command::new("git")
+            .current_dir(self.temp_dir.path())
+            .arg("status")
+            .stdout(std::process::Stdio::inherit())
+            .stderr(std::process::Stdio::inherit())
+            .spawn()
+            .expect("Failed to run git status")
+            .wait()
+            .unwrap();
+
+        assert!(output.success(), "git status failed");
+    }
+
+    pub fn generate_changelog(
+        &self,
+        changelog_config: ChangelogConfig,
+        force_tag: Option<String>,
+    ) -> String {
+        let configure_command = |command: &mut std::process::Command| {
+            command
+                .current_dir(self.temp_dir.path())
+                .arg("--config")
+                .arg(changelog_config.path())
+                .arg("--use-branch-tags")
+                .arg("--unreleased");
+
+            if let Some(tag) = &force_tag {
+                if !tag.contains("-dev") {
+                    command.arg("--tag-pattern").arg("^v\\d+.\\d+.\\d+$");
+                }
+
+                command.arg("--tag").arg(tag);
+            }
+
+            command.arg("--bump");
+        };
+
+        let mut command = std::process::Command::new("git-cliff");
+        configure_command(&mut command);
+
+        if !self.temp_dir.path().join("CHANGELOG.md").exists() {
+            command.arg("--output");
+        } else {
+            command.arg("--prepend").arg("CHANGELOG.md");
+        }
+
+        let exit_status = command
```

Comment:
This is the bit to pay attention to for `git-cliff`. This is what I'd want to codify or put into a GitHub Action script